### ISSUE TITLE
Is the latest release supposed to be stable?

### ISSUE THREAD

**Issue by SgtBlade** [friends-of-forkcms/fork-cms-module-commerce#3]
I've installed this module & followed the instructions but I'm getting errors on a lot of important parts of the module:

Adding new product:
![image](https://user-images.githubusercontent.com/23322965/117276007-e9f15c00-ae5e-11eb-9365-6237a3ee079f.png)

Adding new specification & brand (same code so same error):
![image](https://user-images.githubusercontent.com/23322965/117269078-41d89480-ae58-11eb-89cd-07dc2b3510d2.png)

One of the widgets:
![image](https://user-images.githubusercontent.com/23322965/117269632-be6b7300-ae58-11eb-9bae-d122a70c149e.png)

VAT, Discounts & shop settings work

**Comment by jessedobbelaere** [comment]
hi @SgtBlade , yes this module is still in development, I should mention it in the README.md. I need to use it soon in a project so I plan to finish It though. 

Context: a few agencies using Fork CMS on a daily basis, have their own "webshop" functionality and those are not open-source, unfortunately. After talking to Jacob van Dam a few months ago, I could start from his module https://github.com/jacob-van-dam-ict/fork-cms-catalog and try to make it a full-fledged webshop module for Fork CMS. 

I'm currently making a frontend theme and demo fixtures, and I've set up an automated deploy to a demo environment (https://fork-cms-module-commerce-demo-jessedobbelaere.cloud.okteto.net/). Each PR with new changes also gets deployed so we can preview the changes. I still need to go over the shopping cart and checkout functionality and fix a lot of bugs, write a lot of tests etc. 

(If you're interested in participating to make this module great, feel free to contribute 🙇)

**Comment by SgtBlade** [comment]
@jessedobbelaere Thanks for the quick reply! The way code works in the Fork-CMS is quite different from how I learned it but I'll definitely try and see if I can fix it. I'll post an update here if I found something.

**Comment by jessedobbelaere** [comment]
I just realized I probably had a fix locally for the product bug but did not push it yet. I'll push it

<img width="1414" alt="image" src="https://user-images.githubusercontent.com/1352979/117277079-cda1ef00-ae5f-11eb-9a23-913708ca9f1e.png">
<img width="1416" alt="image" src="https://user-images.githubusercontent.com/1352979/117277100-d266a300-ae5f-11eb-8dd1-f73c4dca5214.png">

**Comment by SgtBlade** [comment]
I've found that you could replace the $event->getData()->id with a random string generator like this:

` MetaType::class,
                    [
                        'base_field_name' => 'title',
                        'generate_url_callback_class' => 'commerce.repository.brand',
                        'generate_url_callback_method' => 'getUrl',
                        'detail_url' => '',
                        'generate_url_callback_parameters' => [
                            $event->getData()->locale,
                            bin2hex(openssl_random_pseudo_bytes(20)),
                        ],
                    ]
`

As far as I've tested, the id being random does not affect anything


https://github.com/friends-of-forkcms/fork-cms-module-commerce/blob/e5bb6180e6e04bc7cb654ba49b132283abfc7ebf/src/Backend/Modules/Commerce/Domain/Brand/BrandType.php#L46

**Comment by jessedobbelaere** [comment]
hi @SgtBlade I spent some time this weekend pushing lots of bugfixes and updates to the codebase. I also pushed a `docs/` folder with some basic info which I'll keep up to date. For now the module is still very much WIP, but there's the demo preview site in the readme to showcase the potential. 

I'm wrapping up the shopping cart functionality with discounts, then doing the checkout process, while creating a demo frontend and fixing bugs/cleaning up the code.

I fixed the bug you mentioned above about the event ID, it was a silly mistake, I simply looked at how other DataTransferObjects work:

https://github.com/friends-of-forkcms/fork-cms-module-commerce/commit/cf6d8f9dd9db9eb55fe95c58dd2ff1954af160d9

Will close the issue for now, I added "WIP - work in progress" in the repo description
